mc/CodeAnalysis/Lexer.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+
+namespace Minsk.CodeAnalysis
+{
+    class Lexer
+    {
+        private readonly string _text;
+        private int _position;
+        private List<string> _diagnostics = new List<string>();
+
+        public Lexer(string text)
+        {
+            _text = text;
+        }
+
+        public IEnumerable<string> Diagnostics => _diagnostics;
+
+        private char Current
+        {
+            get
+            {
+                if (_position >= _text.Length)
```

Comment:
why dont you test for a less than 0 _position here?  are you not worried about an out of range exception?

Reply:
Because I know that never decrement — that would be extremely weird.